xsdata/generators.py

```diff
@@ -70,21 +72,40 @@ def process_class(cls, obj: Class, parents: List[str] = None) -> Class:
 
     @classmethod
     def process_enumerations(cls, obj: Class):
-        values: List[Any] = []
-        for i in range(len(obj.attrs) - 1, -1, -1):
-            attr = obj.attrs[i]
-            cls.process_enumeration(attr, obj)
+        attr_types = {ext.type.name: ext.type for ext in obj.extensions}
+        attrs = {str(attr.default): attr for attr in obj.attrs}
+        obj.attrs = sorted(attrs.values(), key=lambda x: str(x.default))
 
-            if attr.default in values:
-                obj.attrs.pop(i)
-            values.append(attr.default)
+        names = set()
+        for attr in obj.attrs:
+            attr.types.extend(attr_types.values())
+            attr.default = cls.attribute_default(attr)
+            attr.name = cls.enumeration_name(str(attr.default).strip("\"'"))
+            names.add(attr.name)
 
-        obj.attrs.sort(key=lambda x: x.default)
+        if len(names) != len(obj.attrs):
+            for attr in obj.attrs:
+                hash = urlsafe_b64encode(str(attr.default).encode()).decode()
+                attr.name = cls.enumeration_name(hash)
 
     @classmethod
     def process_attributes(cls, obj: Class, parents_list: List[str]):
+        seen: Set[str] = set()
+        obj.attrs = [
+            attr
+            for attr in obj.attrs
+            if attr.name not in seen and seen.add(attr.name) is None  # type: ignore
+        ]
+
+        seen.clear()
         for attr in obj.attrs:
             cls.process_attribute(attr, parents_list)
+            seen.add(attr.name)
+
+        if len(seen) != len(obj.attrs):
+            for attr in obj.attrs:
+                hash = urlsafe_b64encode(str(attr.local_name).encode()).decode()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'hash'](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5070064)